### PR TITLE
Optimize MessageSourceSupport

### PR DESCRIPTION
**Pull request by kang-hl**

### CODE REVIEW

**Comment by qiangmao** [comment]
样式区分

File: spring-context/src/main/java/org/springframework/context/support/MessageSourceSupport.java [hunk 1]
```diff
@@ -116,32 +116,22 @@ protected String formatMessage(String msg, @Nullable Object[] args, Locale local
...
-		MessageFormat messageFormat = null;
-		synchronized (this.messageFormatsPerMessage) {
-			Map<Locale, MessageFormat> messageFormatsPerLocale = this.messageFormatsPerMessage.get(msg);
-			if (messageFormatsPerLocale != null) {
-				messageFormat = messageFormatsPerLocale.get(locale);
+		Map<Locale, MessageFormat> messageFormatsPerLocale = this.messageFormatsPerMessage
+				.computeIfAbsent(msg, key -> new ConcurrentHashMap<>());
+		MessageFormat messageFormat = messageFormatsPerLocale.computeIfAbsent(locale, key -> {
+			try {
+				return createMessageFormat(msg, locale);
```
Comment (qiangmao) [inline spring-context/src/main/java/org/springframework/context/support/MessageSourceSupport.java:123]: 样式区分

Comment (kang-hl) [reply spring-context/src/main/java/org/springframework/context/support/MessageSourceSupport.java:123]: 什么

Comment (kang-hl) [reply spring-context/src/main/java/org/springframework/context/support/MessageSourceSupport.java:123]: > 样式区分

什么

**Comment by snicoll** [comment]
@kang-hl you need to sign the CLA for us to consider this change.

**Pull request closed without merging**